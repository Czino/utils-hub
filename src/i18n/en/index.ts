import date from './date'
import encoding from './encoding'
import form from './form'
import fractions from './fractions'
import home from './home'
import language from './language'
import list from './list'
import navigation from './navigation'
import percentages from './percentages'
import text from './text'
import units from './units'

export default {
  date,
  encoding,
  navigation,
  form,
  fractions,
  home,
  language,
  list,
  units,
  text,
  percentages,
}
