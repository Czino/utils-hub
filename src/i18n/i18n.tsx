import { keys } from '../utils/object/keys'

export const i18n = (template: string, args: Record<string, string> = {}) => {
  let transformed = template

  keys(args).forEach((key) => {
    const value = args[key] || `$${key}`
    transformed = transformed.replace(new RegExp(`\\$${key}`, 'gu'), value)
  })
  return transformed
}

export const i18nElement = (template: string, args: Record<string, string> = {}) =>
  i18n(template, args)
    .split('\n')
    .map((str) => (
      <>
        {str}
        <br />
      </>
    ))
