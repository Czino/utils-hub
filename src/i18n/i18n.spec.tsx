import { i18n, i18nElement } from './i18n'

describe('i18n', () => {
  it('should return the resolved localized text from key with args', () => {
    expect(i18n('this is a localized $foo', { foo: 'bar' })).toBe('this is a localized bar')
    expect(i18n('this is a localized $foo $bar', { foo: 'bar', bar: 'foo' })).toBe('this is a localized bar foo')
    expect(i18n('this is a localized $foo $bar')).toBe('this is a localized $foo $bar')
  })
})
describe('i18nElement', () => {
  it('should return the resolved localized text element from key with args', () => {
    expect(i18nElement('this is a localized\n$foo\n$bar')).toMatchSnapshot('this is a localized $foo $bar')
  })
})
