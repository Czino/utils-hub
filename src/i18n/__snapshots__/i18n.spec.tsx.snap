// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`i18nElement should return the resolved localized text element from key with args: this is a localized $foo $bar 1`] = `
[
  <React.Fragment>
    this is a localized
    <br />
  </React.Fragment>,
  <React.Fragment>
    $foo
    <br />
  </React.Fragment>,
  <React.Fragment>
    $bar
    <br />
  </React.Fragment>,
]
`;
